dbms/src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -240,7 +235,8 @@ class ExpressionAnalyzer : private ExpressionAnalyzerData, private boost::noncop
     size_t subquery_depth;
     bool do_global; /// Do I need to prepare for execution global subqueries when analyzing the query.
 
-    AnalyzedJoin analyzed_join;
+    SyntaxAnalyzerResultPtr syntax;
+    const AnalyzedJoin & analyzed_join;
```

Comment:
const AnalyzedJoin & analyzedJoin() const { return syntax->analyzed_join; } looks better for me